src/main/java/org/jabref/logic/xmp/XMPUtilWriter.java

```diff
@@ -0,0 +1,302 @@
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.strings.StringUtil;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentCatalog;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.pdmodel.common.PDMetadata;
+import org.apache.xmpbox.XMPMetadata;
+import org.apache.xmpbox.schema.DublinCoreSchema;
+import org.apache.xmpbox.xml.XmpSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XMPUtilWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(XMPUtilWriter.class);
+
+    /**
+     * Try to write the given BibTexEntry in the XMP-stream of the given
+     * PDF-file.
+     *
+     * Throws an IOException if the file cannot be read or written, so the user
+     * can remove a lock or cancel the operation.
+     *
+     * The method will overwrite existing BibTeX-XMP-data, but keep other
+     * existing metadata.
+     *
+     * This is a convenience method for writeXMP(File, BibEntry).
+     *
+     * @param filename The filename from which to open the file.
+     * @param entry    The entry to write.
+     * @param database maybenull An optional database which the given bibtex entries belong to, which will be used to
+     *                 resolve strings. If the database is null the strings will not be resolved.
+     * @throws TransformerException If the entry was malformed or unsupported.
+     * @throws IOException          If the file could not be written to or could not be found.
+     */
+    public static void writeXMP(String fileName, BibEntry entry,
+            BibDatabase database, XMPPreferences xmpPreferences) throws IOException, TransformerException {
+        XMPUtilWriter.writeXMP(Paths.get(fileName), entry, database, xmpPreferences);
+    }
+
+    /**
+     * Try to write the given BibTexEntry in the XMP-stream of the given
+     * PDF-file.
+     *
+     * Throws an IOException if the file cannot be read or written, so the user
+     * can remove a lock or cancel the operation.
+     *
+     * The method will overwrite existing BibTeX-XMP-data, but keep other
+     * existing metadata.
+     *
+     * This is a convenience method for writeXMP(File, Collection).
+     *
+     * @param path     The path to write to.
+     * @param entry    The entry to write.
+     * @param database maybenull An optional database which the given bibtex entries belong to, which will be used to
+     *                 resolve strings. If the database is null the strings will not be resolved.
+     * @throws TransformerException If the entry was malformed or unsupported.
+     * @throws IOException          If the file could not be written to or could not be found.
+     */
+    public static void writeXMP(Path file, BibEntry entry,
+            BibDatabase database, XMPPreferences xmpPreferences) throws IOException, TransformerException {
+        List<BibEntry> l = new LinkedList<>();
```

Comment:
Reason for a linked list? In most cases ArrayList is sufficient

Reply:
That's a question, I posted above. This is code from a previous author and I want to drop the List at all. Please comment my question above.